src/storage/core.rs

```diff
@@ -267,6 +267,26 @@ where
         self.write_with_optional_meta(key, value, Some(meta)).await
     }
 
+    /// Free all resources that may be freed without work interruption
+    /// NOTICE! This function frees part of the resources in separate thread,
+    /// so actual resources may be freed later
+    pub async fn free_freeable_resources(&self) {
+        self.observer.try_dump_old_blob_indexes().await;
+        self.close_active_blob_in_background().await;
```

Comment:
This is probably incorrect. We should free only resources, that is optional and was loaded while performing record deletion. Closing an active blob is a visible change and will lead to update of active blob every time, the deletion happened